aten/src/ATen/native/TensorShape.cpp

```diff
@@ -372,6 +370,68 @@ Tensor cat(TensorList tensors, int64_t dim) {
   return result;
 }
 
+Tensor block_diag(TensorList tensors) {
+  Tensor result = at::empty({}, tensors[0].options());
```

Comment:
I realized that this is were my `python test/test_torch.py TestTorchDeviceTypeCPU.test_block_diag_scipy_cpu` failure is coming from. Earlier, I had removed some checks in `block_diag_out` which ensured that all the tensors have the same type, because the block_diag from scipy accepts differing types and just outputs whichever type retains the most information. But I overlooked this `tensor[0].options()` call, which is still setting the type to match the first tensor. I'm not exactly sure at the moment of the best way to implement the behavior I want, so I'll try to figure that out unless pytorch already has a method for that that someone can point me to.

Reply:
Looks like torch.result_type almost does what I need, except that it only operates on a pair of inputs. But I should be able to just call it multiple times to get the behavior I want.